### PR TITLE
Get logs to rollover after a certain time/size

### PR DESCRIPTION
**Pull request by DominicOram**
Fixes DiamondLightSource/python-artemis#482

## To test
* Run for sometime and confirm logs roll over

### CODE REVIEW

File: src/dodal/log.py [hunk 1]
```diff
@@ -21,6 +21,49 @@
...
+                self.stream.seek(0, 2)  # due to non-posix-compliant Windows feature
+                if self.stream.tell() + len(msg) >= self.maxBytes:
+                    return 1
+        return 0
+
+    def getFilesToDelete(self):
+        """
+        Override method to do nothing - we don't want logs to be deleted
+        """
+        return []
```
Comment (DominicOram) [inline src/dodal/log.py:64]: It looks like just setting `backupCount` to 0 will cause no files to ever be deleted. We should do this and remove the override

File: src/dodal/log.py [hunk 2]
```diff
@@ -21,6 +21,49 @@
...
+            return 1
+        else:
+            if self.stream is None:  # delay was set...
+                self.stream = self._open()
+            if self.maxBytes > 0:  # are we rolling over?
+                msg = "%s\n" % self.format(record)
+                self.stream.seek(0, 2)  # due to non-posix-compliant Windows feature
+                if self.stream.tell() + len(msg) >= self.maxBytes:
+                    return 1
+        return 0
```
Comment (DominicOram) [inline src/dodal/log.py:58]: We can simplify this a bit by doing our check first then just returning the result of the super check

File: src/dodal/log.py [hunk 3]
```diff
@@ -76,7 +119,13 @@ def set_up_file_handler(
...
         logging_path = _get_logging_file_path()
         print(f"Logging to {logging_path}")
-    file_handler = logging.FileHandler(filename=logging_path)
+    file_handler = EnhancedRollingFileHandler(
+        filename=logging_path,
+        when="D",
+        interval=1,
+        backupCount=10,
+        maxBytes=1e8,
+    )
```
Comment (DominicOram) [inline src/dodal/log.py:128]: Most of these are the defaults anyway so we can just call it with default args

**Review by olliesilvester — APPROVED** [review]
Looks good - I just changed the name for the test for logs not rolling over

**Pull request merged**